plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/query/CorrelationQueryBuilder.java

```diff
@@ -36,7 +37,7 @@
  *
  * @opensearch.internal
  */
-public class CorrelationQueryBuilder extends AbstractQueryBuilder<CorrelationQueryBuilder> {
+public class CorrelationQueryBuilder extends AbstractQueryBuilder<CorrelationQueryBuilder> implements MultiTermQueryBuilder {
```

Comment:
I will add new interface called `WithFieldName` instead of reusing `MultiTermQueryBuilder`. And rather have `MultiTermQueryBuilder` extend `WithFieldName` in addition to `QueryBuilder` interface